halo2_proofs/src/poly/kzg/commitment.rs

```diff
@@ -132,6 +132,31 @@ where
         }
     }
 
+    /// Initializes parameters for the curve through existing parameters
+    /// k, g, g_lagrange (optional), g2, s_g2
+    pub fn from_params(
```

Comment:
I'll suggest to name it `from_parts`, since the struct's name happens to contain param, which might be confusing.